libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2959,18 +2959,27 @@ void GCS_MAVLINK::handle_common_vision_position_estimate_data(const uint64_t use
                                                               const float roll,
                                                               const float pitch,
                                                               const float yaw,
+                                                              const float covariance[],
                                                               const uint8_t reset_counter,
                                                               const uint16_t payload_size)
 {
 #if HAL_VISUALODOM_ENABLED
+    float posErr = 0;
+    float angErr = 0;
     // correct offboard timestamp to be in local ms since boot
     uint32_t timestamp_ms = correct_offboard_timestamp_usec_to_ms(usec, payload_size);
 
     AP_VisualOdom *visual_odom = AP::visualodom();
     if (visual_odom == nullptr) {
         return;
     }
-    visual_odom->handle_vision_position_estimate(usec, timestamp_ms, x, y, z, roll, pitch, yaw, reset_counter);
+
+    if (!isnan(covariance[0])) {
```

Comment:
Does the covariance appear as a NaN if it's left empty?  I was expecting it would be zero.  No harm in having the NaN check of course.

Reply:
In [mavlink spec](https://mavlink.io/en/messages/common.html#VISION_POSITION_ESTIMATE), it states
>  If unknown, assign NaN value to first element in the array.

Reply:
ah, ok great.